File: TIMS-X.Server/TIMS-X.Server/JsonModels/Patient.ts



    export class Patient { 
        public guid: string;
        public inactive: boolean;
        public salutationId: number | null;
        public providerId: number | null;
        public primaryCareId: number | null;
        public referringPhysicianId: number | null;
        public marketingId: number | null;
        public siteId: number | null;
        public maritalStatusId: string;
        public emplStatusId: string;
        public patientStatusId: number | null;
        public patientTypeId: number | null;
        public otStatusId: number;
        public otStatusDescriptionId: number;
        public birthDate: Date | null;
        public deathDate: Date | null;
        public firstName: string;
        public lastName: string;
        public initial: string;
        public preferredName: string;
        public sex: string;
        public deceased: boolean;
        public email: string;
        public address1: string;
        public address2: string;
        public city: string;
        public state: string;
        public zipCode: string;
        public primaryPhone: PrimaryPhoneEnum;
        public homePhone: string;
        public mobilePhone: string;
        public workPhone: string;
        public otherPhone: string;
        public accountNo: string;
        public notes: string;
        public customText1: string;
        public customText2: string;
        public customDate1: Date | null;
        public customDate2: Date | null;
        public ssn: string;
        public insuredInsurancePayerId: number | null;
        public updatedSiteId: number | null;
        public race: RaceEnum;
        public ethnicity: EthnicityEnum;
        public language: LanguageEnum;
        public legalRep: boolean;
        public legalRepFirstName: string;
        public legalRepLastName: string;
        public legalRepInitial: string;
        public legalRepAddress1: string;
        public legalRepAddress2: string;
        public legalRepCity: string;
        public legalRepState: string;
        public legalRepZipCode: string;
        public legalRepPhone: string;
        public alternateContact: string;
        public alternateContactPhone: string;
        public releaseSignature: boolean;
        public releaseSignatureDate: Date | null;
        public assignBenefits: boolean;
        public assignBenefitsDate: Date | null;
        public responsibleParty: string;
        public hasIntakeData: boolean;
        public lastFirstMiddle: string;
        public firstLast: string;
        public sexFull: string;
        public salutation: Salutation;
        public provider: Provider;
        public primaryCarePhysician: MarketingReference;
        public referringPhysician: MarketingReference;
        public marketing: MarketingReference;
        public updatedByUser: User;
        public createdUserId: number | null;
        public createdDate: Date;
        public privacyDate: Date | null;
        public updatedUserId: number | null;
        public updatedDate: Date;
        public useSecondaryAddress: boolean;
        public secondaryAddress1: string;
        public secondaryAddress2: string;
        public secondaryCity: string;
        public secondaryState: string;
        public secondaryZipCode: string;
        public qbid: string;
        public releaseInformation: boolean;
        public releaseInformationDate: Date | null;
        public consentDate: Date | null;
        public marketingAuthorization: boolean;
        public marketingAuthorizationDate: Date | null;
        public authorizedParties: string;
        public studentStatusId: string;
        public restrictions: PatientRestriction[];
        public medicalConditions: PreviousHistory[];
        public patientTypeIds: number[];
        public restrictionIds: number[];
        public authorizationIds: number[];
        public patientTypeReferences: PatientTypeReference[];
        public authorizationReferences: AuthorizationReference[];
		
        public hasBeenAudited: boolean;
        
        public pendingDelete: boolean;
        public id: number;
        public hasStateBeenSet: boolean;
    }

File: TIMS-X.Core/Domain/Patient.cs
using System;
using System.Collections.Generic;
using TIMS_X.Core.Domain.Base;
using TIMS_X.Core.Enums;

namespace TIMS_X.Core.Domain
{
    public class Patient : AuditableEntity, ICreateByUserAudited, IUpdateAudited, ISiteAuditable, ICreateDateAudited
    {
        public Patient()
        {
            Restrictions = new HashSet<PatientRestriction>();
            MedicalConditions = new HashSet<PreviousHistory>();
            AuthorizationReferences = new HashSet<AuthorizationReference>();
		}

        public Guid Guid { get; set; }
        public bool Inactive { get; set; }
        public int? SalutationId { get; set; }
        public int? ProviderId { get; set; }
        public int? PrimaryCareId { get; set; }
        public int? ReferringPhysicianId { get; set; }
        public int? MarketingId { get; set; }
        public int? SiteId { get; set; }
        public string MaritalStatusId { get; set; }
        public string EmplStatusId { get; set; }
        public int? PatientStatusId { get; set; }
        public int? PatientTypeId { get; set; }
        public int OtStatusId { get; set; }
        public int OtStatusDescriptionId { get; set; }
        public DateTime? BirthDate { get; set; }
        public DateTime? DeathDate { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string Initial { get; set; }
        public string PreferredName { get; set; }
        public string Sex { get; set; }
        public bool Deceased { get; set; }
        public string Email { get; set; }
        public string Address1 { get; set; }
        public string Address2 { get; set; }
        public string City { get; set; }
        public string State { get; set; }
        public string ZipCode { get; set; }
        public PrimaryPhoneEnum PrimaryPhone { get; set; }
        public string HomePhone { get; set; }
        public string MobilePhone { get; set; }
        public string WorkPhone { get; set; }
        public string OtherPhone { get; set; }
        public string AccountNo { get; set; }
        public string Notes { get; set; }
        public string CustomText1 { get; set; }
        public string CustomText2 { get; set; }
        public DateTime? CustomDate1 { get; set; }
        public DateTime? CustomDate2 { get; set; }
        public string Ssn { get; set; }
        public int? InsuredInsurancePayerId { get; set; }
        public int? UpdatedSiteId { get; set; }
        public RaceEnum? Race { get; set; }
        public EthnicityEnum? Ethnicity { get; set; }
        public LanguageEnum? Language { get; set; }
        public bool LegalRep { get; set; } // AKA Name and Address for Invoicing
		public string LegalRepFirstName { get; set; }
        public string LegalRepLastName { get; set; }
        public string LegalRepInitial { get; set; }
        public string LegalRepAddress1 { get; set; }
        public string LegalRepAddress2 { get; set; }
        public string LegalRepCity { get; set; }
        public string LegalRepState { get; set; }
        public string LegalRepZipCode { get; set; }
        public string LegalRepPhone { get; set; }
        public string AlternateContact { get; set; }
        public string AlternateContactPhone { get; set; }
        public bool ReleaseSignature { get; set; }
        public DateTime? ReleaseSignatureDate { get; set; }
        public bool AssignBenefits { get; set; }
        public DateTime? AssignBenefitsDate { get; set; }
        public string ResponsibleParty { get; set; }
        public bool HasIntakeData { get; set; }

        public string LastFirstMiddle => LastName + ", " + FirstName +
                                         (string.IsNullOrEmpty(Initial) ? string.Empty : " " + Initial + ".");
        public string FirstLast => FirstName + " " + LastName;

        public string SexFull
        {
            get
            {
                if (Sex == "M") return "Male";
                if (Sex == "F") return "Female";
                return "Unknown";
            }
        }

        public Salutation Salutation { get; set; }
        public Provider Provider { get; set; }
        public MarketingReference PrimaryCarePhysician { get; set; }
        public MarketingReference ReferringPhysician { get; set; }
        public MarketingReference Marketing { get; set; }
        public User UpdatedByUser { get; set; }
        public int? CreatedUserId { get; set; }
        public DateTime CreatedDate { get; set; }
        public DateTime? PrivacyDate { get; set; }
		public int? UpdatedUserId { get; set; }
        public DateTime UpdatedDate { get; set; }
        public bool UseSecondaryAddress { get; set; }
        public string SecondaryAddress1 { get; set; }
        public string SecondaryAddress2 { get; set; }
        public string SecondaryCity { get; set; }
        public string SecondaryState { get; set; }
        public string SecondaryZipCode { get; set; }
        public string QBID { get; set; }
        public bool ReleaseInformation { get; set; }
        public DateTime? ReleaseInformationDate { get; set; }
        public DateTime? ConsentDate { get; set; }
        public bool MarketingAuthorization { get; set; }
        public DateTime? MarketingAuthorizationDate { get; set; }
        public string AuthorizedParties { get; set; }
        public string StudentStatusId { get; set; }
		public ICollection<PatientRestriction> Restrictions { get; set; }
        public ICollection<PreviousHistory> MedicalConditions { get; set; }
        public int[] PatientTypeIds { get; set; }
        public int[] RestrictionIds { get; set; }
        public int[] AuthorizationIds { get; set; }
        public ICollection<PatientTypeReference> PatientTypeReferences { get; set; }
        public ICollection<AuthorizationReference> AuthorizationReferences { get; set; }

		//public ICollection<PatientInsurance> PatientInsurances { get; set; }

	}
}